Add ThingPulse ePulse Feather and ePulse Feather C6 board definition

squix78 opened the pull request:
Both boards are already on the arduino-esp32 repository:

- https://github.com/espressif/arduino-esp32/blob/master/variants/thingpulse_epulse_feather/pins_arduino.h
- https://github.com/espressif/arduino-esp32/blob/master/variants/thingpulse_epulse_feather_c6/pins_arduino.h

Source:
- https://thingpulse.com/product/epulse-feather-low-power-esp32-development-board/
- https://thingpulse.com/product/epulse-feather-low-power-esp32-c6/

valeros commented on boards/thingpulse_epulse_feather.json:
```diff
@@ -0,0 +1,44 @@
+    },
+    "core": "esp32",
+    "extra_flags": [
+      "-DARDUINO_ESP32_DEV",
```
Shouldn't it be `-DARDUINO_THINGPULSE_EPULSE_FEATHER`?

squix78 replied:
Fixed it

valeros commented on boards/thingpulse_epulse_feather.json:
```diff
@@ -0,0 +1,44 @@
+    "f_flash": "80000000L",
+    "flash_mode": "dio",
+    "mcu": "esp32",
+    "variant": "thingpulse_epulse_feather_c6"
```
Looks like a typo, should be `thingpulse_epulse_feather`.

squix78 replied:
Good catch! Thank you

valeros commented on boards/thingpulse_epulse_feather_c6.json:
```diff
@@ -0,0 +1,30 @@
+    "f_cpu": "160000000L",
+    "f_flash": "80000000L",
+    "flash_mode": "qio",
+    "mcu": "esp32c6",
```
Where are the `extra_flags` and `ldscript` entries for Arduino framework?

squix78 replied:
I added `extra_flags`, but the only example I have for esp32-c6 is the espressif modules and they don't have the `ldscript` value set

valeros commented on boards/thingpulse_epulse_feather_c6.json:
```diff
@@ -0,0 +1,30 @@
+    "variant": "thingpulse_epulse_feather_c6"
+  },
+  "connectivity": [
+    "wifi"
```
Probably `bluetooth` is missing.

squix78 replied:
I added zigbee and thread as well.

valeros commented:
Hi @squix78, thanks for the PR, it will be merged when a new stable release of Arduino core is published. Meanwhile, please address my comments below.

squix78 commented:
Hi @valeros 
Do you feel that I completely addressed your comments? Is the dependency to the arduino framework stable release because I added arduino to supported platforms? I can also remove that now and do another PR once arduino stable is released...

valeros commented:
> Is the dependency to the arduino framework stable release because I added arduino to supported platforms?

That's right, if IDF is enough, you can remove Arduino from the list of supported frameworks.

Jason2866 commented:
@valeros `ldscript` entries for Arduino are not used since core 2.0.0. Can be removed from all board manifest. The usual copy & paste -> it was always there ;-)
